fastcrypto/src/ristretto255.rs

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Implementations of the [ristretto255 group](https://ristretto.group/) which is a group of
+//! prime order 2^{252} + 27742317777372353535851937790883648493.
+
+use std::ops;
+
+use curve25519_dalek_ng;
+use curve25519_dalek_ng::traits::Identity;
+use once_cell::sync::OnceCell;
+use serde::{de, Deserialize, Serialize};
+
+use crate::hash::HashFunction;
+use crate::{error::FastCryptoError, traits::ToFromBytes};
+
+/// Represents a scalar modulo the order of the ristretto group which is 2^{252} + 27742317777372353535851937790883648493.
+pub struct RistrettoScalar(curve25519_dalek_ng::scalar::Scalar);
+
+impl RistrettoScalar {
+    /// Attempt to create a new scalar from the given bytes in canonical representation.
+    pub fn from_canonical_bytes(bytes: [u8; 32]) -> Option<RistrettoScalar> {
+        curve25519_dalek_ng::scalar::Scalar::from_canonical_bytes(bytes).map(RistrettoScalar)
+    }
+
+    /// Create a scalar from the given `u64`.
+    pub fn from(value: u64) -> RistrettoScalar {
+        RistrettoScalar(curve25519_dalek_ng::scalar::Scalar::from(value))
+    }
+
+    /// Create a scalar from the low 255 bits of the given 256-bit integer.
+    pub fn from_bits(value: [u8; 32]) -> RistrettoScalar {
+        RistrettoScalar(curve25519_dalek_ng::scalar::Scalar::from_bits(value))
+    }
```

Comment:
would be nice to add some tests for these methods, qq, do we want to return Option instead of Result?

Reply:
Yes agree with always `returning` a Result for operations that can fail.